openpype/hosts/substancepainter/plugins/publish/validate_ouput_maps.py

```diff
@@ -0,0 +1,110 @@
+    def process(self, instance):
+
+        config = instance.data["exportConfig"]
+
+        # Substance Painter API does not allow to query the actual output maps
+        # it will generate without actually exporting the files. So we try to
+        # generate the smallest size / fastest export as possible
+        config = copy.deepcopy(config)
+        parameters = config["exportParameters"][0]["parameters"]
+        parameters["sizeLog2"] = [1, 1]  # output 2x2 images (smallest)
+        parameters["paddingAlgorithm"] = "passthrough"  # no dilation (faster)
+        parameters["dithering"] = False     # no dithering (faster)
+        config["exportParameters"][0]["parameters"]["sizeLog2"] = [1, 1]
+
+        result = substance_painter.export.export_project_textures(config)
+        if result.status != substance_painter.export.ExportStatus.Success:
+            raise PublishValidationError(
+                "Failed to export texture set: {}".format(result.message)
+            )
+
+        generated_files = set()
+        for texture_maps in result.textures.values():
+            for texture_map in texture_maps:
+                generated_files.add(os.path.normpath(texture_map))
+                # Directly clean up our temporary export
+                os.remove(texture_map)
+
+        creator_attributes = instance.data.get("creator_attributes", {})
+        allow_skipped_maps = creator_attributes.get("allowSkippedMaps", True)
+        error_report_missing = []
+        for image_instance in instance:
+
+            # Confirm whether the instance has its expected files generated.
+            # We assume there's just one representation and that it is
+            # the actual texture representation from the collector.
+            representation = next(iter(image_instance.data["representations"]))
+            staging_dir = representation["stagingDir"]
+            filenames = representation["files"]
+            if not isinstance(filenames, (list, tuple)):
+                # Convert single file to list
+                filenames = [filenames]
+
+            missing = []
+            for filename in filenames:
+                filepath = os.path.join(staging_dir, filename)
+                filepath = os.path.normpath(filepath)
+                if filepath not in generated_files:
+                    self.log.warning(f"Missing texture: {filepath}")
+                    missing.append(filepath)
+
+            if not missing:
+                continue
+
+            if allow_skipped_maps:
+                # TODO: This is changing state on the instance's which
+                #   should not be done during validation.
+                self.log.warning(f"Disabling texture instance: "
+                                 f"{image_instance}")
+                image_instance.data["active"] = False
+                image_instance.data["integrate"] = False
+                representation.setdefault("tags", []).append("delete")
+                continue
```

Comment:
This is disabling maps that do not generate inputs if it's allowed for the instance to ignore those. It's a bit of an odd place for this logic. I'd say it's fine for a merge now since moving it shouldn't break backwards compatibility it would just end up being a bit cleaner.